flink-kubernetes/src/main/java/org/apache/flink/kubernetes/kubeclient/decorators/HadoopConfMountDecorator.java

```diff
@@ -0,0 +1,163 @@
+					localHadoopConfigurationDirectory.get());
+				return flinkPod;
+			}
+
+			final List<KeyToPath> keyToPaths = hadoopConfigurationFileItems.stream()
+				.map(file -> new KeyToPathBuilder()
+					.withKey(file.getName())
+					.withPath(file.getName())
+					.build())
+				.collect(Collectors.toList());
+
+			hadoopConfVolume = new VolumeBuilder()
+				.withName(Constants.HADOOP_CONF_VOLUME)
+				.withNewConfigMap()
+					.withName(getHadoopConfConfigMapName(kubernetesParameters.getClusterId()))
+					.withItems(keyToPaths)
+					.endConfigMap()
+				.build();
+		}
+
+		final Pod podWithHadoopConf = new PodBuilder(flinkPod.getPod())
+			.editOrNewSpec()
+				.addNewVolumeLike(hadoopConfVolume)
+					.endVolume()
+				.endSpec()
+			.build();
+
+		final Container containerWithHadoopConf = new ContainerBuilder(flinkPod.getMainContainer())
+			.addNewVolumeMount()
+				.withName(Constants.HADOOP_CONF_VOLUME)
+				.withMountPath(Constants.HADOOP_CONF_DIR_IN_POD)
+				.endVolumeMount()
+			.addNewEnv()
+				.withName(Constants.ENV_HADOOP_CONF_DIR)
+				.withValue(Constants.HADOOP_CONF_DIR_IN_POD)
+				.endEnv()
+			.build();
+
+		return new FlinkPod.Builder(flinkPod)
+			.withPod(podWithHadoopConf)
+			.withMainContainer(containerWithHadoopConf)
+			.build();
+	}
+
+	@Override
+	public List<HasMetadata> buildAccompanyingKubernetesResources() throws IOException {
+		if (kubernetesParameters.getExistingHadoopConfigurationConfigMap().isPresent()) {
+			return Collections.emptyList();
+		}
+
+		final Optional<String> localHadoopConfigurationDirectory = kubernetesParameters.getLocalHadoopConfigurationDirectory();
+		if (!localHadoopConfigurationDirectory.isPresent()) {
+			return Collections.emptyList();
+		}
+
+		final List<File> hadoopConfigurationFileItems = getHadoopConfigurationFileItems(localHadoopConfigurationDirectory.get());
+		if (hadoopConfigurationFileItems.isEmpty()) {
+			LOG.warn("Found 0 files in directory {}, skip to create the Hadoop Configuration ConfigMap.", localHadoopConfigurationDirectory.get());
+			return Collections.emptyList();
+		}
+
+		final Map<String, String> data = new HashMap<>();
+		for (File file: hadoopConfigurationFileItems) {
+			data.put(file.getName(), Files.toString(file, StandardCharsets.UTF_8));
+		}
+
+		final ConfigMap hadoopConfigMap = new ConfigMapBuilder()
+			.withApiVersion(Constants.API_VERSION)
+			.withNewMetadata()
+				.withName(getHadoopConfConfigMapName(kubernetesParameters.getClusterId()))
+				.withLabels(kubernetesParameters.getCommonLabels())
+				.endMetadata()
+			.addToData(data)
+			.build();
+
+		return Collections.singletonList(hadoopConfigMap);
+	}
+
+	private List<File> getHadoopConfigurationFileItems(String localHadoopConfigurationDirectory) {
+		final File file = new File(localHadoopConfigurationDirectory);
+		if (file.exists() && file.isDirectory()) {
+			return Arrays.stream(file.listFiles()).filter(File::isFile).collect(Collectors.toList());
```

Comment:
You may be right! In addition to the core-site.xml and hdfs-site.xml, I would add core-the default.xml and hdfs-default.xml too.

Reply:
Usually, the `core-default.xml` and `hdfs-default.xml` are bundled in the jar. So it will not exist in the conf directory.

https://www.quora.com/What-is-the-purpose-of-core-default-xml-and-core-site-xml-in-hadoop

Reply:
Sounds reasonable! Remove the default files.